internal/models/psu_bank_bridge_connections.go

```diff
@@ -0,0 +1,240 @@
+	return json.Marshal(&struct {
+		ID                uuid.UUID     `json:"id"`
+		PsuID             uuid.UUID     `json:"psuID"`
+		ConnectorID       string        `json:"connectorID"`
+		CreatedAt         time.Time     `json:"createdAt"`
+		Status            string        `json:"status"`
+		State             CallbackState `json:"state"`
+		ClientRedirectURL *string       `json:"clientRedirectURL,omitempty"`
+		TemporaryToken    *Token        `json:"temporaryToken,omitempty"`
+		Error             *string       `json:"error,omitempty"`
+	}{
+		ID:                a.ID,
+		PsuID:             a.PsuID,
+		ConnectorID:       a.ConnectorID.String(),
+		CreatedAt:         a.CreatedAt,
+		Status:            string(a.Status),
+		State:             a.State,
+		ClientRedirectURL: a.ClientRedirectURL,
+		TemporaryToken:    a.TemporaryToken,
+		Error:             a.Error,
+	})
+}
+
+func (a *PSUBankBridgeConnectionAttempt) UnmarshalJSON(data []byte) error {
+	var aux struct {
+		ID                uuid.UUID     `json:"id"`
+		PsuID             uuid.UUID     `json:"psuID"`
+		ConnectorID       string        `json:"connectorID"`
+		CreatedAt         time.Time     `json:"createdAt"`
+		Status            string        `json:"status"`
+		State             CallbackState `json:"state"`
+		ClientRedirectURL *string       `json:"clientRedirectURL,omitempty"`
+		TemporaryToken    *Token        `json:"temporaryToken,omitempty"`
+		Error             *string       `json:"error,omitempty"`
+	}
+
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	connectorID, err := ConnectorIDFromString(aux.ConnectorID)
+	if err != nil {
+		return err
+	}
+
+	a.ID = aux.ID
+	a.PsuID = aux.PsuID
+	a.ConnectorID = connectorID
+	a.CreatedAt = aux.CreatedAt
+	a.Status = PSUBankBridgeConnectionAttemptStatus(aux.Status)
+	a.State = aux.State
+	a.ClientRedirectURL = aux.ClientRedirectURL
+	a.TemporaryToken = aux.TemporaryToken
+	a.Error = aux.Error
+
+	return nil
+}
+
+// Represents all connections of a psu for a given banking bridge (connector).
+type PSUBankBridge struct {
+	// ID of the connector
+	ConnectorID ConnectorID `json:"connectorID"`
+
+	// Optional
+	// AccessToken is optional for some banking bridges, like Powens, where we
+	// have a notion of connection, but we only have one token for all of them.
+	AccessToken *Token `json:"authToken"`
+	// per banking bridge additional information
+	Metadata map[string]string `json:"metadata"`
+
+	// List of connections for the same banking bridge (for example if the user
+	// connects to multiple banks)
+	Connections []*PSUBankBridgeConnection `json:"connections"`
+}
+
+type ConnectionStatus string
+
+const (
+	ConnectionStatusActive ConnectionStatus = "ACTIVE"
+	ConnectionStatusError  ConnectionStatus = "ERROR"
+)
+
+type PSPPsuBankBridgeConnection struct {
+	// ID of the connection, given by the banking bridge
+	ConnectionID string `json:"connectionID"`
```

Comment:
Maybe should be ConnectionReference as in payment we usually call reference external ids